src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefUserTypeFunctionalTest.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.cwrdapi;
+
+import io.restassured.response.Response;
+import lombok.extern.slf4j.Slf4j;
+import net.serenitybdd.rest.SerenityRest;
+import net.thucydides.core.annotations.WithTag;
+import net.thucydides.core.annotations.WithTags;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.cwrdapi.client.CaseWorkerApiClient;
+import uk.gov.hmcts.reform.cwrdapi.controllers.response.StaffRefDataUserTypesResponse;
+import uk.gov.hmcts.reform.cwrdapi.idam.IdamOpenIdClient;
+import uk.gov.hmcts.reform.cwrdapi.util.FeatureToggleConditionExtension;
+import uk.gov.hmcts.reform.cwrdapi.util.ToggleEnable;
+import uk.gov.hmcts.reform.lib.client.response.S2sClient;
+import uk.gov.hmcts.reform.lib.util.serenity5.SerenityTest;
+
+import javax.annotation.PostConstruct;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.cwrdapi.util.FeatureToggleConditionExtension.getToggledOffMessage;
+
+@ComponentScan("uk.gov.hmcts.reform.cwrdapi")
+@WithTags({@WithTag("testType:Functional")})
+@ActiveProfiles("functional")
+@SerenityTest
+@SpringBootTest
+@Slf4j
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class StaffRefUserTypeFunctionalTest extends AuthorizationFunctionalTest {
+
+    public static final String mapKey = "StaffRefDataController.fetchUserTypes";
+
+    protected static CaseWorkerApiClient caseWorkerApiClient;
+
+    @PostConstruct
+    public void beforeTestClass() {
+
+        SerenityRest.useRelaxedHTTPSValidation();
+
+        if (null == s2sToken) {
+            log.info(":::: Generating S2S Token");
+            s2sToken = new S2sClient(
+                    testConfigProperties.getS2sUrl(),
+                    testConfigProperties.getS2sName(),
+                    testConfigProperties.getS2sSecret())
+                    .signIntoS2S();
+        }
+
+        if (null == idamOpenIdClient) {
+            idamOpenIdClient = new IdamOpenIdClient(testConfigProperties);
+        }
+
+        caseWorkerApiClient = new CaseWorkerApiClient(
+                caseWorkerApiUrl,
+                s2sToken, idamOpenIdClient);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    @ToggleEnable(mapKey = mapKey, withFeature = true)
+    void fetchUserTypeAllStatusCode_200() {
+        StaffRefDataUserTypesResponse responses = (StaffRefDataUserTypesResponse)
+                caseWorkerApiClient.fetchUserType(
+                HttpStatus.OK
+            );
+        assertTrue(responses.getUserTypes().size() > 0);
```

Comment:
1. can you add assertions for expected number of user types and their content.
2. add another test case for empty user types

Reply:
@rajkatla-hmcts - regarding point 1 - don't you think this will be tightly couple with the data? Let say if data will be add / delete / update this case will be fail. Then we need to rewrite the test case. Is this correct approach?

Reply:
@vilasshelke-hmcts ok, ignore first comment

Reply:
point 2 - This case I have handled in JUNIT (StaffRefDataControllerTest.shouldFetchNoUserTypes).